### PR TITLE
if a viral has an install function then call on load!

### PR DESCRIPTION
**Pull request by karlthepagan**
unfortunately this is required to extend most strategy functions with the conventions I've started implementing

I'll try to move away from it, but it's a necessary evil at the moment

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -105,6 +105,9 @@ global.load = (modName) => {
         // load viral overrides 
         mod = infect(mod, 'internalViral', modName);
         mod = infect(mod, 'viral', modName);
+        if( mod.install ) {
```
Comment (Spedwards) [inline main.js]: Check if `mod.install` is also a function.

Comment (karlthepagan) [inline main.js]: I'll throw an error explaining the API contract if it's not a function.

Comment (Spedwards) [inline main.js]: Sounds good.

**Comment by karlthepagan** [comment]
Thinking about the object merge feature I'm going to make this function be the place where that feature could be disabled. Changes still coming, I've had a tough week.

**Comment by Spedwards** [comment]
That actually sounds awesome @karlthepagan.

**Comment by karlthepagan** [comment]
Not done, but getting close.

**Comment by karlthepagan** [comment]
I've been overwhelmed by other systematic changes and I'm abandoning this for the foreseeable future.

**Pull request closed without merging**